includes/wikia/api/ArticlesApiController.class.php

```diff
@@ -776,19 +777,24 @@ protected function getArticlesDetails( $articleIds, $articleKeys = [], $width =
 		$collection = $this->appendMetadata( $collection );
 
 		$thumbnails = null;
-		//if strict return to original ids order
+		$collection = $this->preserveOriginalOrder( $articleIds, $collection );
+		//if strict return to original ids order - as an array
 		if ( $strict ) {
-			foreach( $articleIds as $id ) {
-				if ( !empty( $collection[ $id ] ) ) {
-					$result[] = $collection[ $id ];
-				}
-			}
-			return $result;
+			return array_values( $collection );
 		}
-
 		return $collection;
```

Comment:
Since keys are numeric, could you explain the case on which the return value needs to be different here and in L783?